### PR TITLE
ci: fanout tests

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments

This cuts the time of CI in half, from ~1 hour to ~30 minutes, while adding more robust testing of the `az` and `aws` images (which were previously left out for time savings reasons).  Local `invoke` commands are mostly unchanged, other than the removal of some `pre` invoke commands, but you can now specify a stage (`--stage=[final, minimal, az, aws]`) when performing tasks such as building or testing to support image-level parallelism.  Artifacts are now uploaded individually instead of in groups.

 - [x] Depends on https://github.com/SeisoLLC/easy_infra/pull/143

## Pull Request Checklist

Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [X] Rebase your branch against the latest commit of the target branch
- [X] If you are adding a dependency, please explain how it was chosen
- [X] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [X] If there is an issue associated with your Pull Request, align your PR
  title with [this documentation](https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -242,7 +249,15 @@ def build(_c, debug=False):
             arg = command.upper().replace("-", "_") + "_VERSION"
             buildargs[arg] = CONFIG["commands"][command]["version"]
 
-    for variant in constants.VARIANTS:
+    variants = process_stages(stage=stage)
+
+    for variant in variants:
+        # This pulls the appropriate latest image from Docker Hub to be used as a cache when building
```
Comment (JonZeolla) [inline tasks.py:255]: After my cache testing (commits 3282c53 through 7e27d87 (see [this](https://github.com/SeisoLLC/easy_infra/pull/144/files/2666d69e637c5ce5affc9a14490300353f64b8de..7e27d87741ef558b748dca69ecbd7591229d2b6b) for the end result of those changes)), this appears to be by far the simplest (the cache is automatically updated for each commit to `main` by existing CI workflows) and most robust (works the same in CI and local, and is effective in both) without needing to manage cache hit keys or any additional update/restore logic.

**Comment by JonZeolla** [comment]
Did a review of the changes; rebase/merge deconflict seems to have went well this is good to go

File: tasks.py (hunk 1 above)
Comment (callmestu) [reply tasks.py:255]: After spending a while reading about Github cache hits/misses and restore-key search priority, I'm somewhat surprised you were able to get such a robust solution with no key prefix management!

**Review by callmestu — APPROVED** [review]
Reviewed file changes and conversations. Built locally and tested against .../tests/terraform as per normal. 
CI tests are now parallelized! Speed improvement is remarkable - less than 30 mins! Great!

**Pull request merged**